src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
```

Comment:
For convention, an index variable is usually called `i` or `index`. So, this `d` variable can be renamed to one of those options. Also, the condition of `d < 1000` is not correct, as the divisors array can have a bigger length than this. The correct condition should be `d < divisors.length`, or, after renaming `d` to `i`, `i < divisors.length`.